### DISCUSSION TITLE
Ability to resize UI elements individually (Mainly Chatbox)

### DISCUSSION THREAD

**Discussion by iBrenky** [runelite/runelite#17520] [Ideas]
Being able to resize individual UI elements would be a nice QoL. Having a huge inventory while playing stretched mode is sick. Having a huge chatbox, not so much.

Thank you.

**Comment by DominickCobb-rs** [comment]
https://github.com/runelite/runelite/discussions/17465
https://github.com/runelite/runelite/discussions/16779

**Reply by iBrenky** [reply]
Thank you for listing those. The first one is not really that relevant maybe?

Any logical reason(s), which my stupid ass is probably not seeing, for it not to be worked on? Clearly people want it as a feature ^^

**Reply by DominickCobb-rs** [reply]
AFAIK there isn't an API for directly resizing widgets. I'm unfamiliar with the process of adding to the API, but I would wager it's not easy to add.

e: the stretched mode plugin only handles canvas scaling, and that API just scales the whole canvas.

**Reply by DominickCobb-rs** [reply]
This is technically wrong, there exists some options for widget size in the [widget api](https://static.runelite.net/api/runelite-api/net/runelite/api/widgets/Widget.html) but I'm not versed enough to expand beyond this. Might tinker but also not committing to anything.
